stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,216 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+import regex
+
+
+class InvalidMetadataException(Exception):
+    pass
+
+
+class Connector(BaseJsonSyncConnector):
+    NOZOMI_MAX_PAGE_SIZE = 10000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        return_obj = {}
+        data = []
+        response_dict = {}
+        jwt_token = None
+        local_result_count = 0
+        token_generated = False
+        try:
+            if metadata:
+                if isinstance(metadata, dict) and metadata.get('jwtToken'):
+                    jwt_token = metadata.get('jwtToken')
+                    next_url = metadata.get('next_url')
```

Comment:
This isn't really a url. It's the query portion of the url.

Reply:
Yes, it is not an exact next url; it is a part of a url that contains a query, page number, and page size.
We are keeping this as next_url in the metadata because most of the connectors we used in the metadata dict have this keys result_count, next_url, so keeping the same.

Reply:
So I do get what you are saying, but I believe those other scenarios the next URL is actually given by the API. For example something like

response_data: {next_page_link = "test.com?page=123"}
where the API is directly giving the next url to hit and we just persist it.

The problem here is that I don't believe the API is actually giving us a new URL to use. We are just incrementing the page by 1. In this case, calling it "next_url" could be misleading and give a false impression of how this API works.

In my opinion, I'd just pass along objects in the meta data to represent the parts you need. In this case it'd be "page_number" and "page_size". That said, I left another comment where I think page size should be consistent and should not need to be modified through the queries.

Reply:
Removed next_url in metadata. As per the suggestion using page_size and page_number in metadata.